hvantk/data/file_utils.py

```
@@ -167,3 +171,344 @@ def decompress_files(
+    try:
+        import pysam  # noqa: F401
+
+        return "pysam"
+    except ImportError:
+        return "python"
+
+
+def _convert_with_bgzip(input_path: str, output_path: str, threads: int) -> None:
+    """Convert using system ``bgzip`` via subprocess pipe."""
+    with open(output_path, "wb") as f_out:
+        p1 = subprocess.Popen(
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)